controllers/agents/svc/serviceclass_controller.go

```diff
@@ -200,6 +176,46 @@ func (r *ServiceClassReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	return ctrl.Result{}, errors.Join(errs...)
 }
 
+func Update(ctx context.Context, remote_client client.Client, rs v1alpha1.RegisteredService, secret *v1.Secret) []error {
```

Comment:
as this function is visible in the whole pkg, I'd propose to rename it to `updateRegisteredService`. The same suggestion applies for the above `Delete` function.

Reply:
Fixed.